### PR TITLE
Add the Define Service

### PR DESCRIPTION
**Pull request by ryzokuken**

### CODE REVIEW

File: services/defineService.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+'use strict';
+const request = require('request');
+const constants = require('../constants');
+
+module.exports = {
+  define: (callback, username, term) => {
+    request({
+      url: constants.DEFINE_API_URL + '?term=' + term,
```
Comment (526avijitgupta) [inline services/defineService.js]: Use es6 interpolation syntax ?

``` js
`${constants.DEFINE_API_URL}?term=${term}`,
```

File: services/defineService.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      headers: {
+        "X-Mashape-Key": "MRn5Ke2MMTmshHp839whoCom3Nx2p1Fsdo5jsnBBhSTxU3Zdo2"
+      },
+      method: "GET"
+    }, (error, response, body) => {
+      if (!error && response.statusCode == 200) {
```
Comment (526avijitgupta) [inline services/defineService.js]: Use === instead of ==

File: services/defineService.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        var message = "\n";
```
Comment (526avijitgupta) [inline services/defineService.js]: Use `let` instead of `var`

File: services/defineService.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        body = JSON.parse(body);
```
Comment (526avijitgupta) [inline services/defineService.js]: use `const body = ...`

File: services/defineService.js (hunk 2 above)
Comment (ryzokuken) [reply services/defineService.js]: @526avijitgupta in `response.statusCode == 200` ?

File: services/defineService.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        for(var i = 0; i < body.list.length; i++) {
```
Comment (526avijitgupta) [inline services/defineService.js]: `let i = 0...`

File: services/defineService.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
+'use strict';
+const request = require('request');
+const constants = require('../constants');
+
+module.exports = {
+  define: (callback, username, term) => {
+    request({
+      url: constants.DEFINE_API_URL + '?term=' + term,
+      headers: {
+        "X-Mashape-Key": "MRn5Ke2MMTmshHp839whoCom3Nx2p1Fsdo5jsnBBhSTxU3Zdo2"
```
Comment (yashmehrotra) [inline services/defineService.js]: Add the api key to constants too

File: services/defineService.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+      },
+      method: "GET"
+    }, (error, response, body) => {
+      if (!error && response.statusCode == 200) {
+        var message = "\n";
+        body = JSON.parse(body);
+        for(var i = 0; i < body.list.length; i++) {
+          message += '* ' + body.list[i].definition + '\n';
+          message += 'by **' + body.list[i].author + '** ';
```
Comment (yashmehrotra) [inline services/defineService.js]: use _italics_ for author

File: services/defineService.js (hunk 2 above)
Comment (yashmehrotra) [reply services/defineService.js]: @ryzokuken Yup `===` checks for type also

Comment (yashmehrotra) [reply services/defineService.js]: @ryzokuken basically `200 == "200"` yields true, `200 === "200"` yields false

**Comment by yashmehrotra** [comment]
@ryzokuken Looks good to me.
@526avijitgupta Review, and if ok, merge

File: constants.js [hunk 8]
```diff
@@ -39,5 +43,6 @@ module.exports = {
   SERVER_PREFIX_URL,
   SERVER_DEPLOY_URL,
   SERVER_HOWDOI_PREFIX_URL,
-  JOKES_API_URL
+  JOKES_API_URL,
+  DEFINE_API_URL
```
Comment (526avijitgupta) [inline constants.js]: Where are you exposing `DEFINE_API_KEY` ?

File: services/defineService.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
+'use strict';
+const request = require('request');
+const constants = require('../constants');
+
+module.exports = {
+  define: (callback, username, term) => {
+    request({
+      url: `${constants.DEFINE_API_URL}?term=${term}`,
+      headers: {
+        "X-Mashape-Key": constants.DEFINE_API_KEY
```
Comment (526avijitgupta) [inline services/defineService.js]: `constants.DEFINE_API_KEY` would be undefined.

File: services/defineService.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+      },
+      method: "GET"
+    }, (error, response, body) => {
+      if (!error && response.statusCode === 200) {
+        let message = "\n";
+        const body = JSON.parse(body);
+        for(let i = 0; i < body.list.length; i++) {
```
Comment (526avijitgupta) [inline services/defineService.js]: Can be replaced by 

``` js
body.list.forEach(function(listItem) {
  message += `* ${listItem.definition}\n by *${listItem.author}*  `
  message += `:thumbsup: ${listItem.thumbs_up.toString()} :thumbsdown: ${{listItem.thumbs_down.toString()}\n`;
});
```

^Test once before committing.

File: services/defineService.js (hunk 7 above)
Comment (526avijitgupta) [reply services/defineService.js]: @ryzokuken Resolved this ?

**Comment by 526avijitgupta** [comment]
Great work @ryzokuken 😄 !! Merging this in.

**Pull request merged**